### ISSUE TITLE
multiple ENVs are not set the same way as like single ENV instructions

### ISSUE THREAD

**Issue by childnode** [bmuschko/gradle-docker-plugin#415]
When
```groov
dockerfile(Type:Dockerfile) {
    environmentVariable "foo" "bar"
}
```

we get 
```Dockerfile
ENV FOO bar
```

but with
```groov
dockerfile(Type:Dockerfile) {
    environmentVariable [ "foo":"bar" ]
}
```

we get 
```Dockerfile
ENV "FOO"="bar"
```

This is caused by the [DockerFile.MultiItemJoiner](https://github.com/bmuschko/gradle-docker-plugin/blob/v3.0.11/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy#L510)

referring to https://docs.docker.com/engine/reference/builder/#env and for consistency with the SIngleItemJoiner the code should be
```patch
     static class MultiItemJoiner implements ItemJoiner {
         @Override
         String join(Map map) {
-             map.inject([]) { result, entry -> result << "\"$entry.key\"=\"$entry.value\"" }.join(' ')
+             map.inject([]) { result, entry -> result << "$entry.key=$entry.value" }.join(' ')
         }
     }
```

anyway: this code will (no more) cover spaces in values. but the singleItemJoiner doesn't care about that neither.
both don't care about escaping characters
...another two points we might address

~Marcel

**Comment by orzeh** [comment]
@childnode Thanks for finding this bug! Could you send PR with the fix?

**Comment by childnode** [comment]
just with the line change as described in the patch? sure … next night ;)

**Comment by orzeh** [comment]
@childnode It would be great if you address all issues, i.e. support for spaces in values and escape characters :smile: but lets start with this one line.

**Comment by childnode** [comment]
sorry for the delay, but have had some troubles to get the tests started ... and to get a free evening ;)
so I stumbled over some cases in 
https://docs.docker.com/engine/reference/builder/
1. single `ENV <key> <value>`  instruction
1.a) simple case with a simple word value without any linebreaks. Special Chars (quotes) doesn't matter and are passed as defined
1.b) if value contains a linebreak: ❓undefined if it has to be quoted or (current assumption) one the newline has to be escaped
2. multiple `ENV <key>=<value>[ <key>=<value>]*` instructions behaves a bit different
2.a) simple case with a simple word value without linebreaks. Special Chars (quotes) have to be escaped only if the value is enwrapped in quotes due to spaces
2.b) if value contains a linebreak or space: the value has to be quoted, linebreak is escaped / translated into a single \
3. ❓undefined if keys are allowed to contain whitespaces, I assume not but quotes are allowed
4. this applies to ENV and LABEL

some constraints: if user already quotes the string, we must not repeat this

but, let's see in the PR

**Comment by cdancy** [comment]
Closing as this has been addressed with #419
